desktop/src/com/frostwire/gui/bittorrent/CreateTorrentDialog.java

```diff
@@ -99,6 +96,24 @@
     private String _invalidTrackerURL;
     private JButton _buttonClose;
 
+    private JComboBox _pieceSizeComboBox;
+    private JLabel _labelPieceSize;
+    private int pieceSize;
+    private static final String[] PIECE_SIZE_OPTIONS = {
+            "(Auto Detect)",
+            "16 kB",
+            "32 kB",
+            "64 kB",
+            "128 kB",
+            "256 kB",
+            "512 kB",
+            "1024 kB",
+            "2048 kB",
+            "4096 kB"
+    };
+
+    private static final int[] PIECE_SIZES = {0, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096};
```

Comment:
I think a cooler way of doing this would be to create an `enum`, instead of 2 arrays

```java
private enum PieceSize {
  AUTO_DETECT(0,"(" + I18ntr("Auto Detect") + ")"),
  _16KB(16), // I include the '_' because enums can't start with numbers.
  _32KB(32),
  ...
  _4096KB;

  private int kb;
  private String humanRep;

  PieceSize(int k, String rep) {
      kb = k;
      humanRep = rep;
  }

  PieceSize(int k) {
      this(k, k + " kB");
  } 
}```

Then create the combobox model using the enums.